src/libraries/System.Text.Json/tests/Serialization/Value.ReadTests.NonGenericCollections.cs

```diff
@@ -236,7 +236,7 @@ public static void ReadPrimitiveICollection()
             Assert.Equal(0, count);
         }
 
-        [Fact]
+        [Fact, ActiveIssue("https://github.com/dotnet/runtime/issues/2396")]
```

Comment:
It's a bit unfortunate that a large refactoring PR like this has test side effects like these which makes tracking bug fixes and behavioral differences harder to maintain/find. I would caution against deferring such things to future commits, if possible. If it isn't really practical, then that's fine, but this type of approach tends to atrophy.

Any reason why they can't be fixed here and now (applied generally to all such test changes/modifications)?

Reply:
@Jozkee is taking a look now, and the fix for this issue can be added to this PR.

Reply:
@steveharter, @Jozkee has a fix here: https://github.com/Jozkee/runtime/commit/5a8ea52345bfbfcf859b3d3831e4dc7ee1fb09c9.

Reply:
Yes the issue was there just to unblock this PR since this PR is currently blocking other feature work. I'll try out the fix.